packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts

```diff
@@ -301,6 +301,25 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Ethereum> {
           }
         })
 
+        if (msg.trade) {
```

Comment:
This will send a transaction message for a trade, but there will also be a send and receive message sent for the same transaction based on the current logic.

There is also no `symbol`, `totalValue`, or `token` on the `Trade` payload for sending the `specificTx` function. There will only ever be a single trade payload per transaction as well (no use for Object.entries)

Reply:
ok